File: data_structures/queue/linearqueue_using_array.py
class queue():
	"""docstring for queue"""
	def __init__(self, size=5):
		self.size=size
		self.rear=-1
		self.front=0
		self.arr=size*[None]

	def push(self,value):
		if(self.rear==self.size-1):
			print("queue is full")
			return 0
		else:
			self.rear+=1
			self.arr[self.rear]=value
			print(value, " is added is queue")
			print("front ",self.front)
			print("rear ",self.rear)

	def pop(self):
		if(self.front>self.rear):
			print("queue is empty")
			return 0
		else:
			print(self.arr[self.front], " is poped")
			self.front+=1
			print("front ",self.front)
			print("rear ",self.rear)

	def display(self):
		for i in range(self.front,self.rear+1):
			print(self.arr[i],"\n")


		
list = queue(10)
list.push(5)
list.push(6)
list.push(7)
list.push(54)
list.push(53)
list.push(52)
list.display()
list.pop()
list.pop()
list.pop()
list.display()
list.pop()
list.pop()
list.pop()
list.pop()
list.push(54)
list.push(53)
list.push(52)

File: data_structures/queue/linearqueue_using_array.java
import java.util.*;

public class linearqueue{
	int size,front,rear;
	int arr[];
	public linearqueue(int s){
		size=s;
		front=0;
		rear=-1;
		arr=new int[size];
		System.out.println("created queue of size "+size);
	}
	public int push(int value){
		if(rear==size-1){
			System.out.println(rear+" "+(size-1));
			System.out.println("queue is full");
			return 0;
		}
		else{
			rear++;
			arr[rear]=value;
			System.out.println(value+" is added in queue");
			return 1;
		}
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		linearqueue qu = new linearqueue(10);
		int x = sc.nextInt();
		System.out.println(x);

		
	}

}

